cmd/parca-agent/main.go

```diff
@@ -295,21 +292,46 @@ func main() {
 		})
 	}
 
-	if len(flags.SystemdUnits) > 0 {
+	// Run group for discovery manager
+	var m *discovery.Manager
+	var err error
+
+	{
```

Comment:
Just out of curiosity, why put this code (and that below) in it's own lexical block?

Reply:
Ah, I guess below it scopes `ctx, cancel := context.WithCancel(ctx)` so that makes sense. I don't think there's any such scoping here.

This is definitely just a small nit though, feel free to ignore, I was just curious!

Reply:
Specifically for run groups I kind of like this pattern, but it's certainly debatable. If we want to change that I think we should do that outside of this PR though, as is it's just conforming to the rest of the code.